### ISSUE TITLE
Upgrade to Keycloak 4.2.0.Final

### ISSUE THREAD

**Issue by sebastienblanc** [spring-io/initializr#684]
Keycloak 4.0.0.Final has been released and there also a starter available for Spring Boot 2. I can provide a PR if needed , what I did locally in the application.yml : 
``` 
 mappings:
            - versionRange: 2.0.0.RELEASE
              groupId: org.keycloak
              artifactId: keycloak-spring-boot-2-starter
            - versionRange: "[1.5.3.RELEASE,2.0.0.M1)"
              groupId: org.keycloak
              artifactId: keycloak-spring-boot-starter  
``` 

And the new Keycloak version is updated in the bom section of the yaml file.

**Comment by snicoll** [comment]
Thanks @sebastienblanc, it's a shame we didn't get a chance to know your plan for Keycloak 4 before it went GA. We don't believe that adding the Spring Boot generation in the name of the artifact is a great idea, see [this section of the wiki](https://github.com/spring-projects/spring-boot/wiki/Building-On-Spring-Boot#naming) for more  details.

What are your plans moving forward? Spring Boot 2 support is going to be maintstream eventually so you'll have to remove that "spring-boot-2" starter I suppose, which will increase the maintenance burden on our side.

**Comment by sebastienblanc** [comment]
As you said if SB2 support is going to be mainstream, in the next release of Keycloak we can remove `spring-boot-2-starter` and point the current starter to our SB2 adapter.

**Comment by snicoll** [comment]
@sebastienblanc and I had a discussion offline where I shared that having a starter with the Spring Boot generation in its id was not something we recommend. I very much prefer that users upgrade to a next generation of Keycloak and get the "recommended version" of Spring Boot with it. 

There is no problem, IMO, keeping the previous generation of Keycloak for Spring Boot 1.x (that's what we do for a bunch of dependencies in Spring Boot itself). Having said that, I also think it's sane that a Spring Boot 1 support is still provided, that's what the "legacy" starter arrangement is for [as decribed in the Wiki page](https://github.com/spring-projects/spring-boot/wiki/Building-On-Spring-Boot#naming) I've shared in my previous comment.
